pilot/pkg/model/sidecar.go

```diff
@@ -348,7 +351,15 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *config.Config, config
 				// Default to this hostname in our config namespace
 				if s, ok := ps.ServiceIndex.HostnameAndNamespace[host.Name(h)][configNamespace]; ok {
 					// This won't overwrite hostnames that have already been found eg because they were requested in hosts
-					addService(s)
+					var vss *Service
```

Comment:
nit: seems more readable to have

```
vss := s
```

and I'm not sure if the nil check is necessary? Can we have a nil service in the index?

Reply:
nil check is needed for other case i.e. matchPort is enabled and service does not have port - we just return nil. The same reason I have n't assigned vss to s.